core/src/main/java/org/elasticsearch/index/query/QueryBuilders.java

```diff
@@ -844,6 +850,65 @@ public static PercolatorQueryBuilder percolatorQuery(String documentType, String
                 indexedDocumentRouting, indexedDocumentPreference, indexedDocumentVersion);
     }
 
+    public static void registerBuiltinBuilders(SearchModule searchModule, NamedWriteableRegistry namedWriteableRegistry) {
```

Comment:
I tend to like more the registration as part of SearchModule, I think it belongs there. I see that would require making fromXContent public, but I don't see that as a problem. Otherwise we would need some kind of query registry in this package, but that would be yet another class...I wouldn't go for that.